packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -76,6 +90,43 @@ class HorizontalOverflow extends Foundation<HorizontalOverflowProps,  React.AllH
         this.setState({
             itemsHeight
         });
+
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.addEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentWillUnmount(): void {
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.removeEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * Callback on scroll change
+     */
+    public onScrollChange = (): IScrollChange => {
+        let scrollObject: IScrollChange;
+        const isLtr: boolean = this.getLTR() === Direction.ltr;
+        const distanceRemaining: number =
+            this.horizontalOverflowItemsRef.current.scrollWidth - this.horizontalOverflowItemsRef.current.scrollLeft;
+
+        if (this.horizontalOverflowItemsRef.current.scrollLeft === 0) {
+            scrollObject = { start: isLtr, end: !isLtr };
```

Comment:
you can just execute the props.onScrollChange

Reply:
Not sure I understand based on where this comment is

Reply:
Instead of saving this as a variable you can just execute the this.props.onScrollChange({...data})

Reply:
@janechu are you saying like this?
```
        if (this.horizontalOverflowItemsRef.current.scrollLeft === 0) {
            this.props.onScrollChange({ start: isLtr, end: !isLtr });
        } else if (distanceRemaining === this.horizontalOverflowItemsRef.current.clientWidth) {
            this.props.onScrollChange({ start: !isLtr, end: isLtr });
        } else {
            this.props.onScrollChange({ start: false, end: false });
        }
```

Reply:
yep - that should work

Reply:
@nalogiudice yep

Reply:
Ok